src/gt4py/next/type_system/type_specifications.py

```diff
@@ -92,31 +79,43 @@ def __str__(self) -> str:
         return f"{kind_str}{self.shape}"
 
 
-@dataclass(frozen=True)
-class TupleType(DataType):
-    types: list[DataType]
-
-    def __str__(self) -> str:
-        return f"tuple[{', '.join(map(str, self.types))}]"
+class ListType(DataType):
+    """Represents a neighbor list in the ITIR representation.
 
-    def __iter__(self) -> Iterator[DataType]:
-        yield from self.types
+    Note: not used in the frontend.
```

Comment:
```suggestion
    Note: not used in the frontend. Neighbor fields are represented by local dimensions instead.
```
There was a good reason we put the type here instead of `itir_ts`, but I forgot again. Can we also add comment for the reasoning here?

Reply:
Yes, it appears in FieldType which is the next type below.